### ISSUE TITLE
[C#] Implement new Concept Exercise: attributes

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/csharp#1448]
This issue describes how to implement the `attributes` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Attributes in C#.

## Learning objectives

- Know what attributes are.
- Know the limitations of attributes.
- Know how to add attributes to values.
- Creating a custom attribute.
- Using the `AttributeUsageAttribute` to restrict usage.

## Out of scope

- Using reflection to read attributes.

## Concepts

The Concepts this exercise unlocks are:

- `attributes`: know what attributes are; know the limitations of attributes; know how to add attributes to values; creating a custom attribute.

## Prequisites

This exercise's prerequisites Concepts are:

- `inheritance`: know how to inherit from a class.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Attributes][attributes]: what attributes are and how to use them.
- [Creating custom attributes][custom-attributes]: how to create custom attributes.
- [Restricting attribute usage][attributeusage]: how to restrict attribute usage.

### After

- [Attributes][attributes]: what attributes are and how to use them.
- [Creating custom attributes][custom-attributes]: how to create custom attributes.
- [Restricting attribute usage][attributeusage]: how to restrict attribute usage.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/concepts/pattern_matching.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[attributes]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/
[custom-attributes]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/creating-custom-attributes
[attributeusage]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/attributeusage
